src/components/android-settings-container.js

```diff
@@ -39,12 +44,18 @@ class AndroidSettingsContainer extends React.Component {
      */
     handleChange = e => {
         e.preventDefault()
-        if (e.target.name === 'valuesTEI') {
-            const valueInput = e.target.value
-            e.target.value > maxValues.valuesTEI
-                ? (e.target.value = maxValues.valuesTEI)
-                : (e.target.value = valueInput)
+
+        let { value } = e.target
+        e.target.name === 'encryptDB'
+            ? (value = e.target.checked)
+            : (value = e.target.value)
+
+        if (e.target.name === 'reservedValues') {
+            value > maxValues.reservedValues
```

Comment:
Won't `value` be a string here?  You call `parseInt` on it below, but here you are comparing it to a number

Reply:
I would recommend rewriting this line so that it is simpler, like so:

```js
value = Math.min(maxValues.reservedValues, parseInt(value))
```

Reply:
I rewrote it as you said using Math.min, also I'm assigning value to states:
```
this.setState({
            ...this.state,
            [e.target.name]: value,
        })
```